src/services/offerService.js

```diff
@@ -33,27 +35,35 @@ export const searchOffers = (filters) => async (dispatch) => {
                 error: res.status,
             }));
             dispatch(setLoadingOffers(false));
-            // TODO count metrics
+
+            createEvent(EVENT_TYPES.ERROR(
+                "Offer Search",
+                "BAD_RESPONSE",
+                res.status
+            ));
             return;
         }
         const offers = await res.json();
         dispatch(setSearchOffers(offers.map((offerData) => new Offer(offerData))));
-
         dispatch(setLoadingOffers(false));
-        // TODO count metrics
+
+        recordTime(TIMED_ACTIONS.OFFER_SEARCH, t0, performance.now());
```

Comment:
This is perhaps the most important service on which to collect metrics. It might be interesting to add metrics on which filters are being used, for example to get a feeling of what students look for